### PR TITLE
Mavlink: Implemented SET_CAMERA_ZOOM

### PR DESCRIPTION
**Pull request by dayjaby**
**Describe problem solved by this pull request**
Some cameras, like the FLIR Duo Pro R, don't allow to use the MAV_CMD_SET_CAMERA_ZOOM message to control the zoom. Instead they rely on PWM values.

**Describe your solution**
In mavlink_receiver.cpp we react on MAV_CMD_SET_CAMERA_ZOOM if CAMERA_ZOOM_TYPE is ZOOM_TYPE_RANGE and publish the zoom control in index 4 of actuator_controls_2. This allows maximum flexibility in assigning an AUX channel. Index 4 in control group 2 is still reserved, but seems to be a good spot to place the zoom (index 3 is camera shutter).

**Describe possible alternatives**
One alternative would be to have a parameter similar to MNT_MAN_ROLL/PITCH/YAW where we can set the AUX channel directly. But with the mixer file approach it is possible to allow PWM scaling/reversing.

mavlink_receiver.cpp is probably not the best place to do this implementation. Maybe create an own module called "camera" where we can also handle other CAMERA_ZOOM_TYPEs?

**Test data / coverage**
Only in PX4/sitl_gazebo on the iris with this mixer file: https://gist.github.com/dayjaby/8e451632915cd7c3ec0934ebffca8c52 and adding `set MIXER_AUX mount_zoom` to ROMFS/px4fmu_common/init.d-posix/10016_iris

Doing a test on the real drone today. (Edit: works!)

### CODE REVIEW

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -489,6 +489,28 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
...
+	} else if (cmd_mavlink.command == MAV_CMD_SET_CAMERA_ZOOM) {
+		struct actuator_controls_s actuator_controls = {};
+		actuator_controls.timestamp = hrt_absolute_time();
+
+		for (size_t i = 0; i < 8; i++) {
+			actuator_controls.control[i] = NAN;
+		}
+
+		if ((int)(cmd_mavlink.param1 + 0.5f) == 2) {
+			actuator_controls.control[4] = cmd_mavlink.param2 / 50.0f - 1.0f;
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_receiver.cpp]: Is it documented somewhere that channel 5 is zoom?

Comment (dayjaby) [reply src/modules/mavlink/mavlink_receiver.cpp]: No, channel 5 is still reserved.

Comment (julianoes) [reply src/modules/mavlink/mavlink_receiver.cpp]: Hm, could you add it to the description of the channel then in the message?

Comment (dayjaby) [reply src/modules/mavlink/mavlink_receiver.cpp]: I added a PR to Devguide. Is that what you expected?

Comment (julianoes) [reply src/modules/mavlink/mavlink_receiver.cpp]: And I would add a note or index in the msg file:
https://github.com/PX4/Firmware/blob/master/msg/actuator_controls.msg

**Comment by dayjaby** [comment]
- added enum-like indices to actuator_controls.msg
- allowed DO_CONTROL_VIDEO to appear in missions

File: msg/actuator_controls.msg [hunk 2]
```diff
@@ -9,8 +9,17 @@ uint8 INDEX_FLAPS = 4
...
 uint8 INDEX_LANDING_GEAR = 7
+uint8 INDEX_GIMBAL_SHUTTER = 3
+uint8 INDEX_CAMERA_ZOOM = 4
+
 uint8 GROUP_INDEX_ATTITUDE = 0
 uint8 GROUP_INDEX_ATTITUDE_ALTERNATE = 1
+uint8 GROUP_INDEX_GIMBAL = 2
+uint8 GROUP_INDEX_MANUAL_PASSTHROUGH = 3
+uint8 GROUP_INDEX_MC_VIRTUAL = 4
+uint8 GROUP_INDEX_FW_VIRTUAL = 5
```
Comment (julianoes) [inline msg/actuator_controls.msg]: @sfuhrer are these correct?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 3]
```diff
@@ -489,6 +489,27 @@ void MavlinkReceiver::handle_message_command_both(mavlink_message_t *msg, const
...
+		struct actuator_controls_s actuator_controls = {};
+		actuator_controls.timestamp = hrt_absolute_time();
+
+		for (size_t i = 0; i < 8; i++) {
+			actuator_controls.control[i] = NAN;
+		}
+
+		switch ((int)(cmd_mavlink.param1 + 0.5f)) {
+		case vehicle_command_s::VEHICLE_CAMERA_ZOOM_TYPE_RANGE:
+			actuator_controls.control[actuator_controls_s::INDEX_CAMERA_ZOOM] = cmd_mavlink.param2 / 50.0f - 1.0f;
```
Comment (julianoes) [inline src/modules/mavlink/mavlink_receiver.cpp]: Where do these numbers come from? Are they somehow standardized? In the MAVLink specs it just says:
> The range of valid values depend on the zoom type.
https://mavlink.io/en/messages/common.html#MAV_CMD_SET_CAMERA_ZOOM

Comment (dayjaby) [reply src/modules/mavlink/mavlink_receiver.cpp]: In https://mavlink.io/en/messages/common.html#CAMERA_ZOOM_TYPE it says a value between 0 and 100 is accepted if you have ZOOM_TYPE_RANGE. I map those values to -1 and 1 actuator controls.

File: msg/actuator_controls.msg (hunk 2 above)
Comment (sfuhrer) [reply msg/actuator_controls.msg]: Looking at the guide (https://dev.px4.io/master/en/concept/mixing.html) or [here](https://github.com/PX4/Firmware/blob/abd2bb4eb738eef2007c43206a353ca38d4fdf6e/msg/actuator_controls.msg#L18): yes, ordering is correct. But not sure where we would need the GROUP_INDEX of MC_VIRTUAL and FW_VIRTUAL atm.

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 3 above)
Comment (julianoes) [reply src/modules/mavlink/mavlink_receiver.cpp]: Oh got it!

File: msg/actuator_controls.msg (hunk 2 above)
Comment (hamishwillee) [reply msg/actuator_controls.msg]: @julianoes @dayjaby What needs to happen to progress this? I'm just working through my docs PRs and this is a dependency.

File: msg/actuator_controls.msg [hunk 4]
```diff
@@ -9,8 +9,17 @@ uint8 INDEX_FLAPS = 4
 uint8 INDEX_SPOILERS = 5
 uint8 INDEX_AIRBRAKES = 6
 uint8 INDEX_LANDING_GEAR = 7
+uint8 INDEX_GIMBAL_SHUTTER = 3
+uint8 INDEX_CAMERA_ZOOM = 4
```
Comment (julianoes) [inline msg/actuator_controls.msg]: So these indexes are for another group, right? Otherwise they conflict.

**Comment by julianoes** [comment]
@dayjaby can you make sure this is up-to-date and the CI passes?

**Comment by dayjaby** [comment]
@julianoes Rebased and removed the questionable GROUP_INDEX_MC_VIRTUAL and GROUP_INDEX_FW_VIRTUAL. `make format` tried to change `platforms/common/include/px4_platform_common/i2c_spi_buses.h` in a weird manner, but I just ignored it.

**Pull request merged**